### DISCUSSION TITLE
Gitlab the redirect URI included is not valid.

### DISCUSSION THREAD

**Discussion by softshipper** [nextauthjs/next-auth#1771] [Help] [answered]
Hi all   

Using Gitlab provider I have got the following error message when I try to sign in:
![Screenshot from 2021-04-19 09-17-56](https://user-images.githubusercontent.com/56756949/115196438-306d5980-a0f0-11eb-8ace-9d3573c75b3a.png)

I think the settings should be fine:
![Screenshot from 2021-04-19 09-47-35](https://user-images.githubusercontent.com/56756949/115200269-509f1780-a0f4-11eb-83e8-0967df3b466a.png)

What am I doing wrong?   

Thanks

**Comment by diegoserranor** [comment, marked as answer]
I was having trouble with this too. Setting the **Redirect URI** field to the following did the trick:

`http://localhost:3000/api/auth/callback/gitlab`

**Reply by therealdanmalmx** [reply]
Thanks, saved my day!

**Reply by shpeedle** [reply]
I tried this and it does not work for me. Has something changed?

**Comment by JoNilsson** [comment]
> http://localhost:3000/api/auth/callback/gitlab

Same.  On local dev env, having the redirect URI set to above results in same 404 from GitLab
